rostime/include/ros/time.h

```diff
@@ -151,7 +151,10 @@ namespace ros
 
     double toSec()  const { return (double)sec + 1e-9*(double)nsec; };
     T& fromSec(double t) {
-      sec = (uint32_t)floor(t);
+      int64_t sec64 = (int64_t)floor(t);
+      if (sec64 < 0 || sec64 > UINT_MAX)
+        throw std::runtime_error("Duration is out of dual 32-bit range");
+      sec = (int32_t)sec64;
```

Comment:
Absolutely. Thank you for catching this. To much copy-n-paste between `Duration` and `Time`. I created #63 to address it.